gateways/firstdata/firstdata.go

```diff
@@ -0,0 +1,215 @@
+package firstdata
+
+import (
+	"bytes"
+	"crypto/hmac"
+	"crypto/sha256"
+	"encoding/base64"
+	"encoding/json"
+	"io/ioutil"
+	"net/http"
+	"strconv"
+	"time"
+
+	"github.com/BoltApp/sleet"
+	"github.com/BoltApp/sleet/common"
+)
+
+const (
+	endpoint = "/payments"
+)
+
+type FirstdataClient struct {
+	host            string
+	apiKey          string
+	apiSecret       string
+	clientRequestID string
+	httpClient      *http.Client
+}
+
+func NewClient(env common.Environment, apiKey, apiSecret string) *FirstdataClient {
```

Comment:
Can we actually take in a new struct called `Credentials` that uses apiKey, apiSecret as a string? I have since learned that this is a better pattern because if we have to add variables later on we don't have to change the function signature